src/extension.ts

```diff
@@ -222,27 +222,29 @@ function launchMetals(
       }
     );
 
-    // Open or close the extension output channel. The user may have to trigger
-    // this command twice in case the channel has been focused through another
-    // button. There is no `isFocused` API to check if a channel is focused.
     let channelOpen = false;
-    registerCommand(ClientCommands.TOGGLE_LOGS, () => {
-      if (channelOpen) {
-        client.outputChannel.hide();
-        channelOpen = false;
-      } else {
-        client.outputChannel.show(true);
-        channelOpen = true;
+    const clientCommands: {
+      [k in keyof typeof ClientCommands]: (...args: unknown[]) => unknown
```

Comment:
this is TS voodoo, but in short:

- `{ [k: K]: V }` is basically `Map[K, V]`
- `k in keyof O` means that keys of the Map must be within the keys of object `O`
- `typeof ClientCommands` resolves to `'toggleLogs' | 'focusDiagnostics' | 'runDoctor'` because it's defined as an object literal
- `unknown` is like `any` but it's only a top-type (like `Any` in Scala), so it can't be accidentally used (`any` in TS is also a bottom type, so it basically skips typechecking)